File: app/src/test/java/com/pitecan/gyaim/RomakanaTest.java
//
// ローマ字かな変換のユニットテスト
//
// ここはJavaで書く必要があるのだろうか 2017/07/11 08:56:02
//
package com.pitecan.gyaim;

import org.junit.Test;

import static org.junit.Assert.*;

/**
 * Example local unit test, which will execute on the development machine (host).
 *
 * @see <a href="http://d.android.com/tools/testing">Testing documentation</a>
 */
/*
public class ExampleUnitTest {
    @Test
    public void addition_isCorrect() throws Exception {
        assertEquals(4, 2 + 2);
    }
}
*/

public class RomakanaTest {
    String testdata[][] = {
            {"masui", "ますい", "マスイ"},
            {"dhisukushisutemu", "でぃすくしすてむ", "ディスクシステム"},
            {"vaiorin", "う゛ぁいおりん", "ヴァイオリン"},
            {"hannnya", "はんにゃ", "ハンニャ"}
    };

    @Test
    public void ローマ字変換テスト() throws Exception {
        for (String[] data : testdata) {
            assertTrue(Romakana.roma2hiragana(data[0]).equals(data[1]));
            assertTrue(Romakana.roma2katakana(data[0]).equals(data[2]));
        }
    }
}

File: app/src/test/java/com/pitecan/gyaim/RomakanaTest.kt
//
// ローマ字かな変換のユニットテスト
//
package com.pitecan.gyaim

import org.junit.Test

import org.junit.Assert.*

class RomakanaTest {
    val 変換例リスト = arrayOf(
            arrayOf("masui", "ますい", "マスイ"),
            arrayOf("dhisukushisutemu", "でぃすくしすてむ", "ディスクシステム"),
            arrayOf("dexisukusisutemu", "でぃすくしすてむ", "ディスクシステム"),
            arrayOf("shachou", "しゃちょう", "シャチョウ"),
            arrayOf("syatyou", "しゃちょう", "シャチョウ"),
            arrayOf("vaiorin", "う゛ぁいおりん", "ヴァイオリン"),
            arrayOf("hannnya", "はんにゃ", "ハンニャ"))

    @Test
    fun ローマ字変換テスト() {
        for (変換例 in 変換例リスト) {
            assertTrue(Romakana.roma2hiragana(変換例[0]) == 変換例[1])
            assertTrue(Romakana.roma2katakana(変換例[0]) == 変換例[2])
        }
    }
}
